StatBot/EchoBot.cs

```diff
@@ -23,49 +36,206 @@ public class EchoBot : IBot
         {
             if (context.Activity.Type == ActivityTypes.Message)
             {
-                if (!String.IsNullOrEmpty(context.Activity.Text))
+                var state = ConversationState<Dictionary<string, object>>.Get(context);
+                var dc = dialogs.CreateContext(context, state);
+                await dc.Continue();
+
+                if (!context.Responded)
                 {
-                    var user = DataModel.RememberUser(context.Activity);
-                    List<ITool> _tools;
-                    _tools = new List<ITool>();
-
-                    var baseInterfaceType = typeof(ITool);
-                    var botCommands = Assembly.GetAssembly(baseInterfaceType)
-                        .GetTypes()
-                        .Where(types => types.IsClass && !types.IsAbstract && types.GetInterface("ITool") != null);
-                    foreach (var botCommand in botCommands)
+                    if (context.Activity.Text.ToLowerInvariant().Contains("/createstat"))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS1066)